sources/packages/web/src/components/students/modals/ConfirmEditApplication.vue

```diff
@@ -21,22 +20,42 @@ import { useModalDialog } from "@/composables";
 import { defineComponent } from "vue";
 
 export default defineComponent({
+  props: {
+    beforeEdit: {
+      type: Boolean,
+      required: true,
+    },
+  },
   components: {
     ModalDialogBase,
   },
-  setup() {
+  setup(props) {
     const { showDialog, resolvePromise, showModal } = useModalDialog<boolean>();
     const dialogClosed = () => {
       resolvePromise(false);
     };
     const editApplication = async () => {
       resolvePromise(true);
     };
+
+    let confirmationText = "Submit";
+    let warningText = `Any edits made to your application may result in a new assessment, 
+    potentially delaying your application. Are you sure you want to proceed?`;
+
+    if (props.beforeEdit) {
+      confirmationText = "Edit application";
+      warningText = `Any edits made to your application may require the resubmission of
+      supporting information, potentially delaying your application. Are you
+      sure you want to proceed?`;
+    }
+
```

Comment:
For a logic based reactive property, please use computed.

```ts
const confirmationText = computed(() =>
      props.beforeEdit ? "Edit application" : "Submit",
    );
    const warningText = computed(() =>
      props.beforeEdit
        ? "Any edits made to your application may require the resubmission of supporting information," +
          " potentially delaying your application. Are you sure you want to proceed?"
        : "Any edits made to your application may result in a new assessment," +
          " potentially delaying your application. Are you sure you want to proceed?",
    );
```

Reply:
updated